python/compress-image/test_main.py

```diff
@@ -0,0 +1,359 @@
+        ],
+        [
+            {
+                # Missing api key
+                "payload": {"provider": "krakenio", "image": "12345"},
+                "variables": {"API_KEY": "", "SECRET_API_KEY": ""},
+            },
+            ("Missing API key.")
+        ],
+        [
+            {
+                # Missing secret api key
+                "payload": {"provider": "krakenio", "image": "12345"},
+                "variables": {"API_KEY": "123", "SECRET_API_KEY": ""},
+            },
+            ("Missing api secret key.")
+        ]
+    ])
+    def test_validate_request_ValueError(self, got, want):
+        req = MyRequest({
+            "payload": got["payload"],
+            "variables": got["variables"]
+            })
+        with self.assertRaises(ValueError) as context:
+            main.validate_request(req)
+            self.assertEqual(str(context.exception), want)
+        
+    @parameterized.expand([
+        [
+            {
+                # accessing wrong provider
+                "payload": {"WRONG_PROVIDER": "krakenio", "image": "12345"},
+                "variables": {"API_KEY": "123", "SECRET_API_KEY": ""},
+            }
+        ],
+        [
+            {
+                # accessing wrong image
+                "payload": {"provider": "krakenio", "1Mage": "12345"},
+                "variables": {"API_KEY": "123", "SECRET_API_KEY": ""},
+            }
+        ],
+        [
+            {
+                # accessing wrong api key
+                "payload": {"provider": "krakenio", "1Mage": "12345"},
+                "variables": {"NOT AN API": "123", "SECRET_API_KEY": ""},
+            }
+        ],
+        [
+            {
+                # accessing wrong api keys and secret keys
+                "payload": {"provider": "krakenio", "1Mage": "12345"},
+                "variables": {"API": "123", "SecretKey": ""},
+            }
+        ]
+
+    ])
+    def test_validate_request_KeyError(self, got):
+        req = MyRequest({
+            "payload": got["payload"],
+            "variables": got["variables"]
+            })
+        self.assertRaises(ValueError, main.validate_request, req)
+
+
+class TestMain(unittest.TestCase):
+    @unittest.skipIf(not secret.API_KEY_TINYPNG, "No Tinypng API Key set")
+    def test_main_success(self):
+        # Output validation 1KB
+        want = {
+            "success":
+            True,
+            "image":
+            pathlib.Path(secret.RESULT_1KB_TINYPNG).read_text(encoding="utf-8")
+        }
+
+        req = MyRequest({
+            "payload": {
+                "provider": "tinypng",
+                "image": base64.b64encode(IMAGE_1KB).decode()
+            },
+            "variables": {
+                "API_KEY": secret.API_KEY_TINYPNG
+            }
+        })
+
+        res = MyResponse()  # Create a response object
+        main.main(req, res)
+
+        # Check the response
+        got = res.json()
+        self.maxDiff = None
```

Comment:
Setting an unlimited diff can be expensive when people log test failures and pay for log storage